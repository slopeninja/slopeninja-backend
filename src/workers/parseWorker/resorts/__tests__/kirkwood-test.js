import fs from 'fs';
import {
  parseKirkwoodSnow,
  parseKirkwoodLiftCounts,
  parseKirkwoodLifts,
  parseKirkwoodTrails,
  parseKirkwoodTrailCounts,
} from '../kirkwood';
import { createHtmlParser } from '../../parserFactory';

test('fetches Kirkwood snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/kirkwood-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseKirkwoodSnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: 'Packed Powder',
      newSnow: 4,
      snowDepthBase: null,
      snowDepthSummit: 30,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseKirkwoodSnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Kirkwood lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/kirkwood-lifts.html.input`);
  const resortData = await createHtmlParser('liftCounts', parseKirkwoodLiftCounts)(htmlText);
  expect(resortData).toEqual({
    liftCounts: {
      total: 12,
      open: null,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('liftCounts', parseKirkwoodLiftCounts)('<html></html>');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Kirkwood trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/kirkwood-lifts.html.input`);
  const resortData = await createHtmlParser('trailCounts', parseKirkwoodTrailCounts)(htmlText);
  expect(resortData).toEqual({
    trailCounts: {
      total: 93,
      open: 72,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trailCounts', parseKirkwoodTrailCounts)('<html></html>');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Kirkwood lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/kirkwood-lifts.html.input`);
  const resortData = await createHtmlParser('lifts', parseKirkwoodLifts)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('lifts', parseKirkwoodLifts)('<html></html>');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Kirkwood trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/kirkwood-lifts.html.input`);
  const resortData = await createHtmlParser('trails', parseKirkwoodTrails)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('trails', parseKirkwoodTrails)('<html></html>');
  expect(resortData).toMatchObject({ trails: [] });
});
