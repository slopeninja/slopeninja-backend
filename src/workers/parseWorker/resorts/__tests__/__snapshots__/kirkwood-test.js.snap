// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Kirkwood lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": "Frontside",
      "name": "Snowkirk",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "The Wall",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "The Reut",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "Magic Carpet 2",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "name": "Vista T-Bar",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "Caples Crest",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "name": "Iron Horse",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "name": "Sunrise",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "Solitude",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "Cornice Express",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "TC Express",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "name": "Bunny",
      "status": "closed",
    },
  ],
}
`;

exports[`fetches Kirkwood trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "Backside",
      "level": null,
      "name": "Bud's Alley",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Button Bowl",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Cold Shoulder",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Coyote",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Deadwood Spur",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Devil's Corral",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Devil's Draw",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Easy Rider",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Fawn Ridge",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Happiness is",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Herringbone Straight",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Hully Gully",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Larry's Lip",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Lower Elevator Shaft",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Sunnyside",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "The Wave",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Thimble Peak",
      "status": "closed",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Upper Bud's Alley",
      "status": "open",
    },
    Object {
      "category": "Backside",
      "level": null,
      "name": "Upper Elevator Shaft",
      "status": "closed",
    },
    Object {
      "category": "Bowls",
      "level": null,
      "name": "Eagle Bowl",
      "status": "open",
    },
    Object {
      "category": "Bowls",
      "level": null,
      "name": "Norm's Nose",
      "status": "closed",
    },
    Object {
      "category": "Bowls",
      "level": null,
      "name": "Palisades Bowl",
      "status": "closed",
    },
    Object {
      "category": "Bowls",
      "level": null,
      "name": "Sentinel Bowl",
      "status": "closed",
    },
    Object {
      "category": "Bowls",
      "level": null,
      "name": "Wagon Wheel Bowl",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Bark Shanty",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Bogies Slide",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Boylemaker",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Buckboard",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Chamoix",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Conestoga",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Dick's Drop",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Fireball",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Flying Carpet",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Hay Flat",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Headwaters",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Hell's Delight",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "High Whiskey",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Home Run to Timber Creek",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Jeff's Line",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Jim's",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Juniper",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "KGF",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lightning",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Look Out Janek",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lost Cabin",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Low Whiskey",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lower Caples Crest Traverse",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lower Lookout Janek",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lower Lost Cabin",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lower Monte Wolfe",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lower Olympic",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Lower Zachary",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Mokelumne",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Monte Wolfe",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Olympic",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "One Man Chute",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Outlaw",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Rabbit Runs",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Scott's Run",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Sentinel",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Short Spoke",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Shot 5",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Shotgun",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Snowkirk",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Snowsnake Gully",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Stump Run",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "The Drain",
      "status": "closed",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "The Wall",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Thunder Saddle",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Two Man Chute",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Upper Caples Crest Traverse",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Wagon Trail",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Whiskey Slide",
      "status": "open",
    },
    Object {
      "category": "Frontside",
      "level": null,
      "name": "Zachary",
      "status": "open",
    },
    Object {
      "category": "Terrain Parks",
      "level": null,
      "name": "Buckaroo",
      "status": "closed",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Ditch of Doom",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Free 'N' Easy",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Funny Bunny",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Gold Rush",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Graduation",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Hide N' Seek",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Hole N' Wall",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Jambi's Jaunt",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Jane's Juant",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "School Run",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Squirrel Ridge",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Trench of Terror",
      "status": "open",
    },
    Object {
      "category": "Timber Creek",
      "level": null,
      "name": "Up N' Over",
      "status": "open",
    },
  ],
}
`;
